compiler/src/dotty/tools/dotc/core/TypeErrors.scala

```diff
@@ -140,8 +140,11 @@ class CyclicReference private (val denot: SymDenotation)(using Context) extends
     // cycleSym.flags would try completing denot and would fail, but here we can use flagsUNSAFE to detect flags
     // set by the parser.
     val unsafeFlags = cycleSym.flagsUNSAFE
-    val isMethod = unsafeFlags.is(Method)
+    val isMethod = unsafeFlags.is(Method) // sometimes,isMethod and isConstructor can both be true!
```

Comment:
Yes. It is a method because it is represented with a `def`.

Reply:
if I remember correctly, isMethod is true(which shouldn't be) for case class as in https://github.com/lampepfl/dotty/issues/17076 so I add this warning, maybe I should indicate in a different way?

Reply:
If we print in PostTyper
```scala
        case tree: DefDef =>
          println("---")
          println("DefDef: " + tree.show)
          println("flags: " + tree.symbol.flagsString)
          println("isConstructor: " + tree.symbol.isConstructor)
```
we can see that this class 
```scala
class F(a: Int):
  def this() = this(0)
  def f = 1
```

has the the `Method` flag and `isConstructor` to `true` for both the primary and secondary constructors.

```
---
DefDef: def <init>(a: Int): Unit
flags: <method> <stable> <touched>
isConstructor: true
---
DefDef: def <init>(): Unit =
  {
    this(0)
    ()
  }
flags: <method> <touched> <no-default-params>
isConstructor: true
---
DefDef: def f: Int = 1
flags: <method> <touched>
isConstructor: false
```

Reply:
What might happen is that the `Method` was not in `flagsUNSAFE` because it would be computed and added to `flags` while typing. But the failure may have happened before that.

Reply:
thanks i got it, removed the comment